src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -747,7 +747,8 @@ RouterModifyTaskForShardInterval(Query *originalQuery,
 			continue;
 		}
 
-		shardRestrictionList = make_simple_restrictinfo((Expr *) shardOpExpressions);
+		shardRestrictionList = make_simple_restrictinfo_compat(restriction->plannerInfo,
```

Comment:
Hmm, we synthetically generate `shardOpExpressions`, is it OK to pass `restriction->plannerInfo` here? Shouldn't we pass NULL as we do in the other places? We do not expect any placeholdervars

Reply:
Here it is more about not getting a crash, if plannerInfo is NULL, then make_simple_restrictinfo can crash if there are no placeholdervars and even sending a dummy plannerInfo will prevent that. So I think to be on the safe side, it can be okay to send the current one?

Reply:
> then make_simple_restrictinfo can crash if there are no placeholdervars 

Oh, is it? When I read the code, it seems `root` is only relevant for placeholdervars? 

If that is not the case, how are we passing `NULL` to `pull_varnos()` in this PR?

Reply:
I think I expressed it incorrectly. If there are placeholdervars but we pass NULL as the plannerInfo, then it will crash.

It might be that we won't have placeholdervars here but what I am saying is, there shouldn't be downside of sending a plannerInfo here. Because even if it is not the right plannerInfo, then when that becomes a problem, sending NULL will crash.